src/Symfony/Bridge/Twig/Extension/EmojiExtension.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\Emoji\EmojiTransliterator;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class EmojiExtension extends AbstractExtension
+{
+    private static array $transliterators = [];
+
+    public function __construct(
+        private readonly string $defaultCatalog = 'slack',
+    ) {
+        if (!class_exists(EmojiTransliterator::class)) {
+            throw new \LogicException('You cannot use the "emoji" filter as the "Emoji" component is not installed. Try running "composer require symfony/emoji".');
+        }
+    }
+
+    public function getFilters(): array
+    {
+        return [
+            new TwigFilter('emoji', $this->emoji(...)),
+        ];
+    }
+
+    public function emoji(string $string, ?string $catalog = null): string
```

Comment:
Even if this method is primarily intended for use in a Twig context, it could be helpful to provide a brief docblock describing what it does, including input and output, and maybe a usage example.

I'd find it very useful, so by jumping over the Twig filter in any template, I can quickly read this information without looking at the official documentation.

Reply:
good idea. Done